src/eva/vision/data/datasets/segmentation/bcss.py

```diff
@@ -0,0 +1,240 @@
+"""BCSS dataset."""
+
+import glob
+import os
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Literal, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import torch
+from torchvision import tv_tensors
+from torchvision.transforms.v2 import functional
+from typing_extensions import override
+
+from eva.core.data import splitting
+from eva.vision.data.datasets import _validators, wsi
+from eva.vision.data.datasets.segmentation import _utils, base
+from eva.vision.data.wsi.patching import samplers
+from eva.vision.utils import io
+
+
+class BCSS(wsi.MultiWsiDataset, base.ImageSegmentation):
+    """Dataset class for BCSS semantic segmentation task.
+
+    Source: https://github.com/PathologyDataScience/BCSS
+
+    We apply the the class grouping proposed by the challenge baseline:
+    https://bcsegmentation.grand-challenge.org/Baseline/
+
+    outside_roi: outside_roi
+    tumor: angioinvasion, dcis
+    stroma: stroma
+    inflammatory: lymphocytic_infiltrate, plasma_cells, other_immune_infiltrate
+    necrosis: necrosis_or_debris
+    other: remaining
+
+    Be aware that outside_roi should be assigned zero-weight during model training.
+    """
+
+    _train_split_ratio: float = 0.8
+    """Train split ratio."""
+
+    _val_split_ratio: float = 0.2
+    """Validation split ratio."""
+
+    _expected_length: int = 151
+    """Expected dataset length."""
+
+    def __init__(
+        self,
+        root: str,
+        sampler: samplers.Sampler,
+        split: Literal["train", "val", "trainval", "test"] | None = None,
+        width: int = 224,
+        height: int = 224,
```

Comment:
maybe at a different PR but I would expect these values to be at the sampler's `__init__` values, as I believe this is major part of it (on the class side I think it makes a bit more sense to add it in the init args instead of the `sample` method) - thoughts?

Reply:
Good point. On the class side of things I agree it'd be cleaner to have it in the `sampler` init. But from a user perspective, having it exposed at the dataset init might be easier to understand / more intuitive.

No strong preference though, we can implement this in a follow up PR.

Reply:
Opened an issue for this here: https://github.com/kaiko-ai/eva/issues/568